File: LeetCode_1365/Main.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
    const arr = [...nums];
    arr.sort((a, b) => a - b);

    const cnt = new Array(101).fill(-1);

    for (let i = 0; i < arr.length; i++) {
        if (cnt[arr[i]] == -1) {
            cnt[arr[i]] = i;
        }
    }
    const answer = [];
    for (let num of nums) {
        answer.push(cnt[num]);
    }
    return answer;
};

console.log(smallerNumbersThanCurrent(nums = [8,1,2,2,3]));

File: LeetCode_1365/Main.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> arr = nums;
        sort(arr.begin(), arr.end());

        vector<int> cnt(101, -1);
        for (int i = 0; i < arr.size(); i++) {
            if (cnt[arr[i]] == -1) {
                cnt[arr[i]] = i;
            }
        }

        vector<int> answer;
        for (int num : nums) {
            answer.push_back(cnt[num]);
        }
        return answer;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {8, 1, 2, 2, 3};
    vector<int> ans = sol.smallerNumbersThanCurrent(nums);

    for (int x : ans) cout << x << " ";
    cout << endl;

    return 0;
}
